apps/system/js/app_usage_metrics.js

```diff
@@ -785,6 +812,24 @@
     }
   };
 
+  UsageData.prototype.recordSearch = function(provider) {
+    debug('recordSearch', provider);
+
+    if(provider == null) {
+      return;
+    }
+
+    // We don't want to report search metrics for local search and any other
+    // situation where we might be offline.  Check this here as this may change
+    // in the future.
+    if(navigator.onLine) {
+      var search = this.getSearchCounts(provider);
+      search.count++;
+      debug('Search Count : ', search.count);
```

Comment:
nit: add the provider to this debug statement so we can monitor counts for each?